com.unity.perception/Runtime/GroundTruth/Labelers/KeypointLabeler.cs

```diff
@@ -108,14 +107,62 @@ bool AreEqual(Color32 lhs, Color32 rhs)
             return lhs.r == rhs.r && lhs.g == rhs.g && lhs.b == rhs.b && lhs.a == rhs.a;
         }
 
+        bool PixelOnScreen(int x, int y, (int x, int y) dimensions)
+        {
+            return x >= 0 && x < dimensions.x && y >= 0 && y < dimensions.y;
+        }
+
+        bool PixelsMatch(int x, int y, Color32 idColor, (int x, int y) dimensions, NativeArray<Color32> data)
+        {
+            var h = dimensions.y - y;
+            var pixelColor = data[h * dimensions.x + x];
+            return AreEqual(pixelColor, idColor);
+        }
+
+        static int s_PixelTolerance = 2;
```

Comment:
Can you explain why we need '2' here? a 5x5 kernel is a lot of ops, and it feels like we might be covering up a bug in some other math here.

Reply:
yep, needed to round instead of just casting.